tests/modules/testrdb.c

```diff
@@ -313,6 +325,16 @@ int RedisModule_OnLoad(RedisModuleCtx *ctx, RedisModuleString **argv, int argc)
             return REDISMODULE_ERR;
     }
 
+    /* Used to verify that aux_save2 api without any data, saves nothing to the RDB. */
+    RedisModuleTypeMethods datatype_methods = {
+        .version = REDISMODULE_TYPE_METHOD_VERSION,
+        .aux_load = test2rdb_aux_load,
+        .aux_save2 = test2rdb_aux_save,
+        .aux_save_triggers = REDISMODULE_AUX_BEFORE_RDB | REDISMODULE_AUX_AFTER_RDB,
+    };
+
+    RedisModule_CreateDataType(ctx, "test2_rdb", 1, &datatype_methods);
```

Comment:
maybe it's nicer to add this to the else-if above and depend on the startup arguments.
i.e. the module always has only one data type with different behaviors depending on the argument.
maybe we should also have a variant of `aux_save` (1) that doesn't save anything, and a variant of `aux_save2` who does?